File: src/main/java/xyz/upperlevel/hgame/world/entity/EntitySpawnPacket.java
package xyz.upperlevel.hgame.scenario.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;
import xyz.upperlevel.hgame.network.Packet;
import xyz.upperlevel.hgame.network.ProtocolId;


@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString
@ProtocolId(0)
public class EntitySpawnPacket implements Packet {
    private int entityTypeId;
    private float x, y;
    private boolean isFacingLeft;
}

File: src/main/kotlin/xyz/upperlevel/hgame/world/entity/EntitySpawnPacket.kt
package xyz.upperlevel.hgame.world.entity

import xyz.upperlevel.hgame.network.Packet
import xyz.upperlevel.hgame.network.ProtocolId


@ProtocolId(0xE0_0001)
open class EntitySpawnPacket(var entityTypeId: String,
                             var entityId: Int,
                             var x: Float,
                             var y: Float,
                             var isFacingLeft: Boolean) : Packet

@ProtocolId(0xE0_0002)
class ThrowableEntitySpawnPacket(entityTypeId: String,
                                 entityId: Int,
                                 x: Float,
                                 y: Float,
                                 isFacingLeft: Boolean,
                                 var throwerEntityId: Int) : EntitySpawnPacket(entityTypeId, entityId, x, y, isFacingLeft)
